### ISSUE TITLE
Incorrect diagnostic augmentation for `{{yield}}` type errors

### ISSUE THREAD

**Issue by dfreeman** [typed-ember/glint#375]
We see a TS2345 on a top-level mustache and automatically assume its an attempt to emit invalid content, but `{{yield}}` is a special form that we should special-case.

```
Only primitive values and certain DOM objects (see `ContentValue` in `@glint/template`) are usable as top-level template content.
  Argument of type '[]' is not assignable to parameter of type 'NonNullable<FlattenBlockParams<ExpandSignature<T>["Blocks"]>["default"]>'.glint:ts(2345)
```
